File: 240608/3n + 1 수열 m번/3n-plus-1-sequence-m-times.py
M = int(input())
n_lst = [int(input()) for _ in range(M)]

for n in n_lst:
    cnts = 0
    while n != 1:
        if n % 2 == 0:
            n //= 2
        else:
            n *= 3
            n += 1
        
        cnts += 1
    
    print(cnts)

File: 251109/3N + 1 수열 M번/3n-plus-1-sequence-m-times.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() 
{
    // Please write your code here.
    int m, num;
    cin >> m;
    vector<long long> n_lst(m);

    for(int i = 0; i < m; i++)
    {
        cin >> n_lst[i];
    }

    for(long long n: n_lst)
    {
        int cnts = 0;
        while(n != 1)
        {
            if(n % 2 == 0)
                n /= 2;
            else
                n = n * 3 + 1;

            cnts++;
        }
        cout << cnts << endl;
    }

    return 0;
}